vpr/src/device/rr_graph_view.h

```diff
@@ -96,6 +96,60 @@ class RRGraphView {
         return node_storage_.fan_in(node);
     }
 
+    /* Get the xlow of a routing resource node. This function is inlined for runtime optimization. */
+    inline short node_xlow(RRNodeId node) const {
+        return node_storage_.node_xlow(node);
+    }
+
+    /* Get the xhigh of a routing resource node. This function is inlined for runtime optimization. */
+    inline short node_xhigh(RRNodeId node) const {
+        return node_storage_.node_xhigh(node);
+    }
+
+    /* Get the ylow of a routing resource node. This function is inlined for runtime optimization. */
+    inline short node_ylow(RRNodeId node) const {
+        return node_storage_.node_ylow(node);
+    }
+
+    /* Get the yhigh of a routing resource node. This function is inlined for runtime optimization. */
+    inline short node_yhigh(RRNodeId node) const {
+        return node_storage_.node_yhigh(node);
+    }
+
+    /* Get the length of a routing resource node. This function is inlined for runtime optimization. */
+    inline int node_length(RRNodeId node) const {
+        return 1 + node_xhigh(node) - node_xlow(node) + node_yhigh(node) - node_ylow(node);
+    }
+
+    /* Check if routing resource node is initialized. This function is inlined for runtime optimization. */
+    inline bool node_is_initialized(RRNodeId node) const {
```

Comment:
This API is confusing, which could be misleading. 
- If we want to check a node is initialized or not, it should be limited to the ``SOURCE`` type. If a node is uninitialized, it is more nature to be set its type to ``NUM_RR_TYPE``, rather than ``SOURCE``.
- The all-0 ``xlow/ylow/xhigh/yhigh`` do exist, for example, a PLL may locate at ``(0, 0)``, which definitely has a SOURCE node which is all-0 in coordinates. As for me, the uninitialized coordinates should all ``-1``.

Therefore, this API is quite confusing to me. See if @vaughnbetz has more insights on this.